src/exception/repo/entity.ts

```diff
@@ -0,0 +1,132 @@
+      submitter_specimen_id: String,
+    },
+  ],
+  follow_up: [
+    {
+      ...BaseExceptionSchema.obj,
+      submitter_followup_id: String,
+    },
+  ],
+});
+
+// check if model exists already to account for file watchers eg. test runner with live reload
+const EntityExceptionModel =
+  mongoose.models.EntityException ||
+  mongoose.model<EntityException>('EntityException', entityExceptionSchema);
+
+const entityExceptionRepository = {
+  async save(exception: OnlyRequired<EntityException, 'programId'>): Promise<EntityException> {
+    L.debug(`Creating new donor exception with: ${JSON.stringify(exception)}`);
+
+    const update = { $set: exception };
+
+    try {
+      const doc = await EntityExceptionModel.findOneAndUpdate(
+        { programId: exception.programId },
+        update,
+        { upsert: true, new: true, returnDocument: 'after' },
+      ).lean(true);
+      return doc;
+    } catch (e) {
+      L.error('failed to create entity exception: ', e);
+      throw new DatabaseError('Cannot save entity exception.');
+    }
+  },
+
+  async find(programId: string): Promise<EntityException | null> {
+    L.debug(`finding entity exception with program id: ${JSON.stringify(programId)}`);
+    try {
+      // first found document, or null
+      const doc = await EntityExceptionModel.findOne({ programId }).lean(true);
+      return doc;
+    } catch (e) {
+      L.error('failed to find program exception', e);
+      throw new DatabaseError();
+    }
+  },
+
+  async deleteSingleEntity(
+    programId: string,
+    entity: Entity,
+    submitterDonorIds: string[],
+  ): Promise<EntityException | null> {
+    L.debug(
+      `deleting single entity ${entity} exception with program id: ${JSON.stringify(programId)}`,
+    );
+    try {
+      const entityExceptionDoc = await EntityExceptionModel.findOne({ programId });
+      if (entityExceptionDoc) {
+        /**
+         * typescript union array methods don't work well particulary pre v4 (currently on 3.9.5)
+         * all our entity types union with BaseEntityExceptionRecord
+         * filter only uses the `submitter_donor_id` field which is in BaseEntityExceptionRecord
+         * explicitly adding `any` typings so it's very obvious we loose type data here
+         */
+        const entitiesToFilter: any = entityExceptionDoc[entity];
+        const filteredEntities = entitiesToFilter.filter(
+          (entity: BaseEntityExceptionRecord) =>
+            !submitterDonorIds.includes(entity.submitter_donor_id),
+        );
+        entityExceptionDoc[entity] = filteredEntities as any;
+        const doc = await entityExceptionDoc.save();
+        return doc;
+      }
+      // mongo will return nulls for non existent docs
+      // tslint doesn't complain about a null from a lib
+      // tslint:disable-next-line
+      return null;
+    } catch (e) {
+      L.error('failed to delete exception', e);
+      throw new DatabaseError('Cannot save entity exception.');
```

Comment:
Ok. maybe this was on purpose, but just in case...
```suggestion
      throw new DatabaseError('Cannot delete entity exception.');
```